service/routes.py

```diff
@@ -162,3 +161,27 @@ def check_content_type(media_type):
         status.HTTP_415_UNSUPPORTED_MEDIA_TYPE,
         "Content-Type must be {}".format(media_type),
     )
+
+######################################################################
+#  Increment Interested Counter
+######################################################################
+
+
+@app.route('/recommendations/<int:rec_id>/interested', methods=["PUT"])
+def increment_interested_counter(rec_id):
+    """
+    Increment a recommendation's interesed field
+    This endpoint will increment the interested counter by one when interested button is clicked
+    """
+    app.logger.info(
+        'Increment intrested field for recommendation with id: %s', rec_id)
+    check_content_type('application/json')
+    recommendation = Recommendation.find(rec_id)
+    if not recommendation:
+        raise NotFound(
+            "Recommendation with id '{}' was not found.".format(rec_id))
+    count = recommendation.rec_interested
+    if count == None:
+        count = 0
+    recommendation.rec_interested = count + 1
```

Comment:
Can you help me understand where/how the counter is incremented in the DB?

Reply:
In DB, I added a rec_interested field (you may see change in model.py), initial value is None. When "interested" is clicked for the first time, the count is set to 0, and will be increase by 1. Line 184 and line 185  are used to parse the value of count from None to an integer 0 so that it can be increased by 1.